readme.md

```diff
@@ -103,22 +119,38 @@ The extension MUST only inject itself into pages having the `window.kilt` object
 } as InjectedWindowProvider;
 ```
 
-The extension SHOULD perform the following tasks in `signWithDid` and `signExtrinsicWithDid`:
+The extension SHOULD perform the following tasks in `getSignedDidCreationExtinsic`, `signWithDid` and `signExtrinsicWithDid`:
 - protect against Denial-of-Service attacks where the dApp floods the extension with requests
 - ensure that the user has previously authorized interaction with this dApp
 - otherwise, request user authorization for this interaction
 
 
 ## Cryptography
 
-The signing is done using an authorization key of a DID.
+The signing of `getSignedDidCreationExtinsic` is done using a keypair.
+The allowed types for these keys are sr25519 and ed25519.
+
+The signing of `signWithDid` and `signExtrinsicWithDid` is done using an authorization key of a DID.
 The allowed types for these keys are sr25519, ed25519, and ecdsa.
 
 * The type sr25519 uses Schnorr signature scheme with Curve25519 keypair.
 * The type ed25519 uses Ed25519 signature scheme with Curve25519 keypair.
 * The type ecdsa uses ECDSA signature scheme with secp256k1 curve keypair and blake2 as the hashing algorithm.
 
 
+## DID Creation
+
+1. The dApp calls `getSignedDidCreationExtinsic` passing the submitter as a parameter.
+2. The extension presents an interface for generating and signing the on-chain DID creation extrinsic.
+    It SHOULD allow the user to choose the keypair for signing.
+    It SHOULD allow the user to reject the creation and signing of the extrinsic.
+3. The extension generates and signs the on-chain DID creation extrinsic
+    It MUST include an authentication key.
+    It MAY include other keys and services.
+4. The extension resolves the promise returned from `getSignedDidCreationExtinsic` with the signed extrinsic.
+5. The dApp submits the extrinsic or sends it to the the submitter it specified in the API call.
```

Comment:
I’d rather not include this point at all. This is already beyond the scope of the API we’re describing.

Reply:
I meant point 5:
>5. The dApp submits the extrinsic or sends it to the the submitter it specified in the API call.